content/en/blog/istio-traffic-management-walkthrough/index.md

```diff
@@ -0,0 +1,1162 @@
+
+To allow incoming traffic to reach a service running inside the cluster, you need to create an external load balancer first. As part of the installation, Istio deploys an `istio-ingressgateway` service that is of type `LoadBalancer`. Together with this service and an Istio Gateway resource, you can configure access to services running inside the cluster. 
+
+If you run `kubectl get svc istio-ingressgateway -n istio-system`, you will get an output similar to this one: 
+
+~~~~~~~~sh
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP       PORT(S)...
+istio-ingressgateway   LoadBalancer   10.107.249.46   <pending>         80:31380/TCP... 
+~~~~~~~~
+
+The above output shows the Istio ingress gateway of type LoadBalancer. If you're using a Minikube cluster you will notice how the external IP column shows text `<pending>` - that is because we don't have a real external load balancer as everything runs locally. With a cluster running in the cloud from any cloud provider, we would see an actual IP address there - that IP address is where the incoming traffic enters the cluster. If you are using Docker for Mac/Windows, you will see `localhost` under the EXTERNAL-IP column.
+
+We will be accessing the service in the cluster frequently, so we need to know which address to use. The address we are going to use depends on where the Kubernetes cluster is running. 
+
+### If using Minikube
+
+Minikube has a command called `minikube tunnel`. This command creates networking routes from your machine into the Kubernetes cluster as well as allocates IPs to services marked with LoadBalancer. What this means is that you can access your exposed service using an external IP address, just like you would when you're running Kubernetes in the cloud. 
+
+To use the tunnel command, open a new terminal window and run `minikube tunnel`. You should see an output similar to this one: 
+
+~~~~~~~~yaml
+$ minikube tunnel
+Status:
+    machine: minikube
+    pid: 43606
+    route: 10.96.0.0/12 -> 192.168.99.104
+    minikube: Running
+    services: [istio-ingressgateway]
+  errors:
+    minikube: no errors
+    router: no errors
+    loadbalancer emulator: no errors
+~~~~~~~~
+
+If you run the `kubectl get svc istio-ingressgateway -n istio-system` command to get the ingress gateway service, you will notice an actual IP address in the `EXTERNAL-IP` column. It should look something like this: 
+
+~~~~~~~~sh
```

Comment:
does `~~~~~~~sh` render well in Hugo?

Reply:
Yes, but just to be on the safe side, I'll modify it.